query-engine/query-engine-wasm/build.sh

```diff
@@ -29,7 +29,7 @@ then
 fi
 
 echo "Building query-engine-wasm using $WASM_BUILD_PROFILE profile"
-wasm-pack build "--$WASM_BUILD_PROFILE" --target $OUT_TARGET --out-name query_engine
+CARGO_PROFILE_RELEASE_OPT_LEVEL="z" wasm-pack build "--$WASM_BUILD_PROFILE" --target $OUT_TARGET --out-name query_engine
```

Comment:
`CARGO_PROFILE_RELEASE_OPT_LEVEL="z"` is equivalent to:

```
# Cargo.toml

[profile.release]
opt-level = 'z'
```

Let's stick with the current convention we've already adopted :)

Reply:
Also, there's `wasm-opt -Oz`, which "super-focuses on code size"

Reply:
@jkomyno adding this to `[profile.release]` will affect the native build as well. The proper way to do that via `Cargo.toml` would be to add a new profile that inherits from `release` (e.g. `wasm-release` as I did in https://github.com/prisma/prisma-engines/pull/4449), change the nix derivation to use `wasm-release` instead of `release` and pass `--profile=$WASM_BUILD_PROFILE` rather than just `--$WASM_BUILD_PROFILE` to wasm-pack (as the shortcut is only available for `release` and `dev`).

Reply:
Good point Alexey, and I like the dedicate build profile idea.
In retrospect, applying compile-time env vars directly like Miguel did is probably the easiest solution.
It's a matter of deciding where do we want to look for the set build configurations. Arguably, centralising them makes it easier to find them and change them as needed

Reply:
@aqrln is right, I applied the configuration as an env var because otherwise it would affect other targets. I think the profile idea is more involved, but I'm fine centralizing around profiles rather than env. (Although a new profile, will require it's corresponding env toggle too)

Reply:
I think a dedicated profile is still the only correct solution, using an env var like this sort of breaks the proper debug builds. Also there's an unaddressed comment from Alberto about missing `-Os` to `-Oz` change in `wasm-opt` flags. It's nothing critical, we can do both changes in follow-up PRs, just mentioning it for the record.